connectors/uplynk/src/main/java/com/theoplayer/android/connector/uplynk/internal/UplynkAdIntegration.kt

```diff
@@ -1,62 +1,221 @@
 package com.theoplayer.android.connector.uplynk.internal
 
 import com.theoplayer.android.api.THEOplayerView
+import com.theoplayer.android.api.ads.Ad
 import com.theoplayer.android.api.ads.ServerSideAdIntegrationController
 import com.theoplayer.android.api.ads.ServerSideAdIntegrationHandler
+import com.theoplayer.android.api.event.player.PlayerEvent
 import com.theoplayer.android.api.event.player.PlayerEventTypes
+import com.theoplayer.android.api.event.player.SeekedEvent
+import com.theoplayer.android.api.event.player.TimeUpdateEvent
 import com.theoplayer.android.api.player.Player
 import com.theoplayer.android.api.source.SourceDescription
 import com.theoplayer.android.api.source.drm.DRMConfiguration
 import com.theoplayer.android.api.source.drm.KeySystemConfiguration
+import com.theoplayer.android.connector.uplynk.SkippedAdStrategy
 import com.theoplayer.android.connector.uplynk.UplynkAssetType
+import com.theoplayer.android.connector.uplynk.UplynkConfiguration
 import com.theoplayer.android.connector.uplynk.UplynkSsaiDescription
 import com.theoplayer.android.connector.uplynk.internal.network.PreplayInternalLiveResponse
 import com.theoplayer.android.connector.uplynk.internal.network.PreplayInternalVodResponse
 import com.theoplayer.android.connector.uplynk.internal.network.UplynkApi
+import kotlin.time.Duration
 import kotlin.time.DurationUnit
 import kotlin.time.toDuration
 
+private enum class State {
+    PLAYING_CONTENT,
+    SKIPPING_TO_AD_BREAK,
+    SKIPPED_TO_AD_BREAK,
+    PLAYING_SKIPPED_AD_BREAK,
+    FINISHED_PLAYING_SKIPPED_AD_BREAK,
+}
+
+private class SeekTime {
+    companion object {
+        private val DEFAULT_VALUE = 0.0.toDuration(DurationUnit.SECONDS)
```

Comment:
```suggestion
        private val DEFAULT_VALUE = Duration.ZERO
```
You might also want to remove this property and just use `Duration.ZERO` everywhere.